core/src/main/java/com/linecorp/armeria/common/AggregatedHttpRequest.java

```diff
@@ -0,0 +1,251 @@
+     */
+    static AggregatedHttpRequest of(HttpMethod method, String path, MediaType mediaType,
+                                    String format, Object... args) {
+        requireNonNull(method, "method");
+        requireNonNull(path, "path");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(format, "format");
+        requireNonNull(args, "args");
+        return of(method, path, mediaType,
+                  HttpData.of(mediaType.charset().orElse(StandardCharsets.UTF_8), format, args));
+    }
+
+    /**
+     * Creates a new HTTP request.
+     *
+     * @param method the HTTP method of the request
+     * @param path the path of the request
+     * @param mediaType the {@link MediaType} of the request content
+     * @param content the content of the request
+     */
+    static AggregatedHttpRequest of(HttpMethod method, String path, MediaType mediaType, byte[] content) {
+        requireNonNull(method, "method");
+        requireNonNull(path, "path");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        return of(method, path, mediaType, HttpData.of(content));
+    }
+
+    /**
+     * Creates a new HTTP request.
+     *
+     * @param method the HTTP method of the request
+     * @param path the path of the request
+     * @param mediaType the {@link MediaType} of the request content
+     * @param content the content of the request
+     * @param offset the start offset of {@code content}
+     * @param length the length of {@code content}
+     */
+    static AggregatedHttpRequest of(
+            HttpMethod method, String path, MediaType mediaType, byte[] content, int offset, int length) {
+        requireNonNull(method, "method");
+        requireNonNull(path, "path");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        return of(method, path, mediaType, HttpData.of(content, offset, length));
+    }
+
+    /**
+     * Creates a new HTTP request.
+     *
+     * @param method the HTTP method of the request
+     * @param path the path of the request
+     * @param mediaType the {@link MediaType} of the request content
+     * @param content the content of the request
+     */
+    static AggregatedHttpRequest of(HttpMethod method, String path, MediaType mediaType, HttpData content) {
+        requireNonNull(method, "method");
+        requireNonNull(path, "path");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        return of(method, path, mediaType, content, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP request.
+     *
+     * @param method the HTTP method of the request
+     * @param path the path of the request
+     * @param mediaType the {@link MediaType} of the request content
+     * @param content the content of the request
+     * @param trailingHeaders the trailing HTTP headers
```

Comment:
Trailers can only be in the response, so we don't need them in the API for `AggregatedHttpRequest`.

Another benefit of splitting the classes :D

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer

Reply:
Isn't it possible to have trailers in the request over HTTP/2 connection?

https://httpwg.org/specs/rfc7540.html#HttpSequence

Reply:
Ah thanks! Then we do need them. I mistakenly thought HTTP/2 had the same semantics as HTTP/1 for trailers orz